ingester/ingester.go

```diff
@@ -515,7 +515,7 @@ func (i *Ingester) flushSeries(u *userState, fp model.Fingerprint, series *memor
 	}
 
 	firstTime := series.chunkDescs[0].FirstTime()
-	flush := immediate || model.Now().Sub(firstTime) > i.cfg.MaxChunkAge
+	flush := immediate || len(series.chunkDescs) > 1 || model.Now().Sub(firstTime) > i.cfg.MaxChunkAge
```

Comment:
More of less.  Series.Add will 'close' chunks when they get full:

https://github.com/weaveworks/cortex/blob/master/ingester/series.go#L141

(chunks don't actually have a closed concept, we just track it for the head chunk specially)